tests/ui/option_map_unit_fn.stderr

```diff
@@ -0,0 +1,156 @@
+error: called `map(f)` on an Option value where `f` is a unit function
+  --> $DIR/option_map_unit_fn.rs:33:5
+   |
+33 |     x.field.map(do_nothing);
+   |     ^^^^^^^^^^^^^^^^^^^^^^^-
+   |     |
+   |     help: try this: `if let Some(...) = x.field { do_nothing(...) }`
+   |
+   = note: `-D option-map-unit-fn` implied by `-D warnings`
+
+error: called `map(f)` on an Option value where `f` is a unit function
+  --> $DIR/option_map_unit_fn.rs:35:5
+   |
+35 |     x.field.map(do_nothing);
+   |     ^^^^^^^^^^^^^^^^^^^^^^^-
+   |     |
+   |     help: try this: `if let Some(...) = x.field { do_nothing(...) }`
```

Comment:
Maybe we could generate a variable name? The name could be taken from the variable that is matched on.

Reply:
Something like `x_field` or `field` in this case? I guess it could cause problems (with rustfix?) if there is already a variable with the same name as the generated one?

Reply:
There could be, but let's do it anyway and mark the suggestion as approximate